lighthouse-core/gather/gatherers/start-url.js

```diff
@@ -49,24 +49,29 @@ class StartUrl extends Gatherer {
   pass(options) {
     return options.driver.getAppManifest()
       .then(response => {
-        return manifestParser(response.data, response.url, options.url);
+        return response && manifestParser(response.data, response.url, options.url);
       })
       .then(manifest => {
-        if (!manifest.value.start_url || !manifest.value.start_url.raw) {
-          return Promise.reject(new Error(`No web app manifest found on page ${options.url}`));
+        if (!manifest || !manifest.value.start_url || !manifest.value.start_url.raw) {
+          throw new Error(`No web app manifest found on page ${options.url}`);
         }
 
         if (manifest.value.start_url.debugString) {
-          return Promise.reject(new Error(manifest.value.start_url.debugString));
+          throw new Error(manifest.value.start_url.debugString);
         }
 
         this.startUrl = manifest.value.start_url.value;
-      }).then(_ => this.executeFetchRequest(options.driver, this.startUrl));
+      })
+      .then(_ => this.executeFetchRequest(options.driver, this.startUrl))
+      .catch(err => {
+        this.debugString = err.message;
```

Comment:
probably better to set `this.debugString` above and `return` rather than throwing errors and catching everything here. Doing that would also let errors from `getAppManifest`, the manifest parser, and `executeFetchRequest` get through without all being turned into debugStrings